pkg/pki/pgp/pgp.go

```diff
@@ -277,3 +277,17 @@ func (k PublicKey) KeyRing() (openpgp.KeyRing, error) {
 
 	return k.key, nil
 }
+
+// EmailAddresses implements the pki.PublicKey interface
+func (k PublicKey) EmailAddresses() []string {
+	var names []string
+	// Extract from cert
+	for _, entity := range k.key {
+		for _, identity := range entity.Identities {
+			if len(identity.UserId.Email) > 0 {
```

Comment:
should we do any checking here for valid formatting?

Reply:
Yeah, looks like it's not enforced https://www.rfc-editor.org/rfc/rfc4880.html#section-5.11